### ISSUE TITLE
Current states (xx Tagen) als offline Gerät

### ISSUE THREAD

**Issue by Nogger82** [ciddi89/ioBroker.device-watcher#31]
Ich lasse mir die Meldung via Telegram senden. Folgendes wird gesendet:

_Folgende 2 Geräte sind seit einiger Zeit nicht erreichbar:
Current states (39 Tagen)
Temperatursensor Garage (48 Stunden)_

Woher kommt "Current states"? Kann man den ausblenden?

Vielen Dank für den Adapter.

**Comment by ciddi89** [comment]
Da scheint er irgendwie den Namen falsch zu beziehen. Kannst du mir sagen von welchen Adapter er das current states zieht? Sollte dann ja in der JSON Tabelle offline devices zu sehen sein. Ich danke schon mal

**Comment by Nogger82** [comment]
Dort bekomme ich folgendes
[{"Device":"Current states","Adapter":"Nuki-extended","Last contact":"39 Tagen"}
Also der Nuki-extended. Aber da gibt es doch nur den reinen ID "State" Ordner mit dem Namen "Current states"

**Comment by Nogger82** [comment]
Kann ich helfen? Vielleicht irgendwelche Objekte auslesen oder so?

**Comment by ciddi89** [comment]
Soweit ich weiß gibt es nicht viele Objekte für Nuki Geräte. Aber ein Screenshot oder die JSON der objektstruktur hilft vielleicht schon.

**Comment by Nogger82** [comment]
> Soweit ich weiß gibt es nicht viele Objekte für Nuki Geräte. Aber ein Screenshot oder die JSON der objektstruktur hilft vielleicht schon.

Wenn du mir jetzt noch erklärst, wie ich die JSON raus lasse, kann ich das gerne tun

**Comment by ciddi89** [comment]
Den Ordner auswählen (in dem Fall den Ordner von Nuki) und dann auf das rot umkreiste Download Symbol. Dann auf nur ausgewählte. Dann bekommst du eine TXT Datei wo der JSON Code drin liegt. Aber ein Screenshot würde auch reichen. :) 

![image](https://user-images.githubusercontent.com/66088020/184002246-5898f10d-82fd-4b4b-ad79-ba02e172997f.jpeg)

**Comment by ciddi89** [comment]
Ich denke ich habe es schon herausgefunden. Habe nun eingestellt das er vom Hauptordner den Namen ziehen soll. Bitte testen ob es auch funktioniert. Über die JSON würde ich mich trotzdem freuen um zu sehen welche Datenpunkte ich da sonst noch nutzen kann. Denke BatteryCritical, den ich gerade nutze, ist für die Zeitangabe wann sich das Gerät zu letzt gemeldet hat schlecht gewählt.

**Comment by Nogger82** [comment]
> Ich denke ich habe es schon herausgefunden. Habe nun eingestellt das er vom Hauptordner den Namen ziehen soll. Bitte testen ob es auch funktioniert. Über die JSON würde ich mich trotzdem freuen um zu sehen welche Datenpunkte ich da sonst noch nutzen kann. Denke BatteryCritical, den ich gerade nutze, ist für di
[nuki-extended.0.smartlocks.Musterhaus123.State.zip](https://github.com/ciddi89/ioBroker.device-watcher/files/9307339/nuki-extended.0.smartlocks.Musterhaus123.State.zip)
e Zeitangabe wann sich das Gerät zu letzt gemeldet hat schlecht gewählt.

Ich hänge mal die JSON der States an.

**Comment by Nogger82** [comment]
Habe nun auf die 0.3.0 geupdatet. Jetzt steht statt Current states der Name des Nuki. Ich hoffe du bekommst was aus der JSON raus. Die aktuellsten States sind bei mir "lastDataUpdate" und "lastStateUpdate". Vielleicht hilft das ja. Wenn man natürlich längere Zeit nicht die Tür bedient oder ein Schließvorgang durch Zeitsteuerung erfolgt, dann wird das wahrscheinlich auch kein Update erfolgen.

**Comment by ciddi89** [comment]
Habe nun mal als Hauptdatenpunkt lastDataUpdated genommen. Vielleicht zeigt er die Zeit nun besser an? Zusätzlich sollte auch die Battery angezeigt werden falls das Gerät diesen Wert liefert.

**Comment by Nogger82** [comment]
Also aktuell hat sich noch nichts geändert oder muss ich noch ein Update fahren?
Ansonsten wäre der Wert für die Batterie ".state.batteryChargeState". Eine Volt Angabe habe ich nicht gefunden.

**Comment by ciddi89** [comment]
Ja einfach nochmal neu drüber installieren

**Comment by Nogger82** [comment]
Also, update habe ich gemacht.

Der Nuki wird nun als Name angezeigt und der letzte Stand ist heute (13.08.2022) "Last contact":"11:53 Uhr". Er ist aber ganz normal online und der letzte Stand im "lastupdated" ist auch 11:53 Uhr von heute.
Der Witz ist, ich habe gestern um 13:30 mal die Batterie aus einem Xiaomi Tempsensor entfernt. Letzte Änderung der Temperatur "12.08.22 um 13:23 Uhr". Das scheint der Adapter aber irgendwie nicht mitzubekommen. 
Einstellung im Adapter für die Offlinezeiten Zigbee war nicht "-1" sondern Testweise "1", habe ich nun wieder auf 300 gestellt.
Das einzige "Offline" Geräte ist der Nuki. Mache ich irgendwas flasch?

**Comment by ciddi89** [comment]
Also verstehe ich das richtig das der Adapter den Nuki trotzdem als offline anzeigt? Was ist denn eingetragen bei offline Zeit vom Nuki in den Experten Einstellungen? Mit letzte Änderungen der Temperatur kann der Adapter auch nicht mitbekommen da er nicht nach den Datenpunkt schaut.

**Comment by Nogger82** [comment]
> Also verstehe ich das richtig das der Adapter den Nuki trotzdem als offline anzeigt? Was ist denn eingetragen bei offline Zeit vom Nuki in den Experten Einstellungen? Mit letzte Änderungen der Temperatur kann der Adapter auch nicht mitbekommen da er nicht nach den Datenpunkt schaut.

Ja, der Nuki wird stets als Offline gemeldet. Ich lasse ihn jeden Abend um 22 Uhr abschließen, da bekomme ich gestern abend um 23:15 Uhr vom device-watcher die Meldung, dass der Nuki seit 22 Uhr nicht mehr erreichbar sei. Wo kann ich denn die Offline Zeit vom Nuki sehen? 
Den  Temperatursensor, dem ich am 12.08. die Batterie gezogen habe, hat er mir aber gestern dann auch endlich angezeigt, mit 33 Stunden offline. 

Zusammenfassend, bei jeder Telegrammeldung vom Device-Watcher ist der Nuki als offline gemeldet.

**Comment by ciddi89** [comment]
In den Instanz Einstellungen vom Device Einstellungen und Experten Einstellungen steht das vielleicht auf `-1` ? Ändere das Bitte mal auf zb. 300. Muss das -1 bei Nuki Devices rausnehmen da die kein erreichbar Datenpunkt haben.

<img width="1061" alt="Screenshot 2022-08-15 at 10 53 02" src="https://user-images.githubusercontent.com/66088020/184606496-70307bb1-4dcc-422c-98f2-e3c40d766e54.png">

**Comment by Nogger82** [comment]
Alles klar, habe ich jetzt mal umgesetzt. Danke dir für den Tipp.

**Comment by ciddi89** [comment]
@Nogger82 Und funktioniert das jetzt? :)

**Comment by Nogger82** [comment]
> @Nogger82 Und funktioniert das jetzt? :)

Jap läuft, Dankeschön 😊

**Comment by ciddi89** [comment]
Super, dann kann ich das ja schließen. Falls noch was ist einfach melden. 😊

**Comment by Nogger82** [comment]
Schlechte Nachrichten, die Meldung ist wieder da. Einmal letzte Nacht um 3:15 wurde nuki seit 5 Stunden offline gemeldet und dann heute mittag um 12  Uhr, auch wieder für 5 Stunden.

**Comment by ciddi89** [comment]
Das ist ja auch richtig so. Dann wurde 5 Stunden der Datenpunkt nicht aktualisiert vom Nuki Adapter/Gerät. Dann eventuell die dreihundert Minuten(5 std) erhöhen. Ich weiß leider nicht wie das bei Nuki Geräten ist. Bei anderen Geräten werden meistens die Datenpunkte zumindest alle paar Minuten/Stunden aktualisiert. Daher gibt es die Möglichkeiten mit der Feineinstellungen in den Experten Einstellungen da es überall anders ist.

**Comment by Nogger82** [comment]
Okay, ich stell mal 12 Stunden ein, das muss dann passen, weil in diesem Zeitraum eigentlich immer was passiert.

**Comment by ciddi89** [comment]
Wäre natürlich besser wenn es da einen Datenpunkt gibt der sich zwischen durch aktualisiert wenn das Schloss erreichbar ist, ohne das man das Schloss ausgelöst hat. Vielleicht macht der das ja auch, nur im größeren Zeitraum. Muss man mal beobachten. (Noch) habe ich kein Nuki Schloss. Dann ist das immer schwierig über Umwege rauszufinden. 🙈 Ich danke dir trotzdem für deine Geduld und Information! :)

**Comment by Nogger82** [comment]
Sieht eher schlecht aus. Ich habe mir mal alle Datenpunkte des Nuki angeschaut, da ändert sich nichts, bos wirklich eine Änderung am Schloß passiert. Eine Art lifebit gibt es leider auch nicht. Mit den zwölf Stunden kommt man im Alltag gut hin. Wochendenden und Urlaub könnten ea dabei dann wieder raus reißen denke ich.
